engine/src/main/scala/cromwell/engine/backend/BackendConfiguration.scala

```diff
@@ -1,25 +1,48 @@
 package cromwell.engine.backend
 
 import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.backend.{BackendConfigurationDescriptor, BackendLifecycleActorFactory}
+import cromwell.engine.CromwellConfiguration
 
 import scala.collection.JavaConverters._
+import scala.util.{Try, Success, Failure}
 
-/**
-  * Defines a backend configuration.
-  *
-  * @param name      Backend name.
-  * @param className Implementation class for the specific backend.
-  */
-case class BackendConfigurationEntry(name: String, className: String, config: Config)
+case class BackendConfigurationEntry(name: String, className: String, shadowLifecycleActorFactoryClass: String, config: Config) {
+  def asBackendLifecycleActorFactory: BackendLifecycleActorFactory = {
+    Class.forName(shadowLifecycleActorFactoryClass)
+         .getConstructor(classOf[Config])
+         .newInstance(config)
+         .asInstanceOf[BackendLifecycleActorFactory]
+  }
+
+  val asBackendConfigurationDescriptor = BackendConfigurationDescriptor(config, CromwellConfiguration.config)
+}
 
 object BackendConfiguration {
-  private val BackendConfig = ConfigFactory.load().getConfig("backend")
+  private val BackendConfig = CromwellConfiguration.config.getConfig("backend")
   private val DefaultBackendName = BackendConfig.getString("default")
-  private val BackendProvidersConfigs = BackendConfig.getConfigList("providers").asScala.toList
+  private val BackendProviders = BackendConfig.getConfig("providers")
+  private val BackendNames: Set[String] = BackendProviders.entrySet().asScala.map(_.getKey.split("\\.").toSeq.head).toSet
+
+  val AllBackendEntries: List[BackendConfigurationEntry] = BackendNames.toList map { backendName =>
+    val entry = BackendProviders.getConfig(backendName)
+    // TODO PBE: "class" can be removed when switching to the shadow world; "actor-factor" renamed?
```

Comment:
for real or sarcastic?  I don't mind it I guess, I didn't have it in me to think of a better name at the time.